python/mpt/tests/src/mpt/test_trie_diff.py

```diff
@@ -62,6 +65,94 @@ def embedded_node_strategy(draw):
     }
 
 
+@composite
+def invalid_mpt_strategy(draw):
+    """
```

Comment:
I think we should rather split this into multiple strategies and have one test per strategy rather than rely on hypothesis to auto-generate all cases we want to cover.

Also, after we split in multiple test, we can target a specific error the test shuold raise in the pytest.raise section of the test.

Reply:
how important is it?

Reply:
quite important imo as this is typically the kind of strategy where we expect each case to be covered explicitly and raise its own error message

Reply:
> quite important imo as this is typically the kind of strategy where we expect each case to be covered explicitly and raise its own error message

made a proposal